Fatras/include/ActsFatras/Digitization/detail/ParametersSmearer.hpp

```diff
@@ -0,0 +1,96 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2020 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#pragma once
+
+#include <utility>
+
+#include "Acts/Utilities/Definitions.hpp"
+#include "Acts/Utilities/ParameterDefinitions.hpp"
+#include "Acts/Utilities/Result.hpp"
+
+namespace ActsFatras {
+
+/// Smearing functions definition:
+/// - it takes the unsmeared parameter
+/// - it returns the smeared parameter and a covariance
+using SmearFunction =
+    std::function<Acts::Result<std::pair<double, double>>(double)>;
+
+namespace detail {
+
+/// @brief single component smearing struct
+///
+/// This is used to unpack the parameter pack smearing
+/// @param idx The index of the smearing
+/// @param[in,out] values The vector of values to be smeared
+/// @param[in,out] covariances The covariance matrix to be set
+/// @param sFunction The smearing function to be used
+/// @param[in,out] result a recursive smearing result trigger
+///        if a single component is not ok, this will be communicated
+///        upstream to the caller
+template <typename values_t, typename covariances_t>
```

Comment:
```suggestion
template <typename values_t, typename covariance_t>
```
Nitpicking: covariance does not need to be plural as it alway refers to at least two variables.

Reply:
Changed.